sn_networking/src/record_store.rs

```diff
@@ -208,40 +241,29 @@ impl RecordStore for DiskBackedRecordStore {
         Self::read_from_disk(k, &self.config.storage_dir)
     }
 
-    fn put(&mut self, r: Record) -> Result<()> {
-        trace!("PUT request for Record key: {:?}", r.key);
-        if r.value.len() >= self.config.max_value_bytes {
-            warn!(
-                "Record not stored. Value too large: {} bytes",
-                r.value.len()
-            );
-            return Err(Error::ValueTooLarge);
+    fn put(&mut self, record: Record) -> Result<()> {
+        if self.records.contains(&record.key) {
+            trace!("Unverified Record {:?} already exists.", record.key);
+            return Ok(());
```

Comment:
We might want to send out an event even if we already have it? E.g. a double spend attempt would trigger a PUT with the same record key.

Reply:
considering there is `double spend ` case, then it will be better just throw it to the validation and leave it to be detected and handled there.